lib/lcdbwf/tests/testthat/test-results.R

```diff
@@ -0,0 +1,139 @@
+#library(testthat)
+#library(DESeq2)
+#library(lcdbwf)
+#library(rlang)
+#library(stringr)
+#library(BiocParallel)
+#library(ggplot2)
+#library(AnnotationHub)
+#library(dplyr)
+#register(MulticoreParam(config$parallel$cores))
+#devtools::load_all('../../../../lib/lcdbwf')
+source('test-functions.R')
+config <- lcdbwf:::load_config('../../../../workflows/rnaseq/downstream/config.yaml')
+text <- yaml::yaml.load_file('../../../../workflows/rnaseq/downstream/text.yaml')
+
+# Wrapper function for inherits ggplot
+is_ggplot <- function(x) {
+  inherits(x, "ggplot")
+}
+
+# Function to capture mdcat output
+mdcat_output <- c()
+mock_mdcat <- function(...) {
+  mdcat_output <<- c(mdcat_output, paste(..., collapse = " "))
+}
+
+# Helper function to create DESeqResults object
+make_deseq_results <- function(test='Wald', type='ashr', reduced_design=NULL) {
+  design_data <- make_design_data()
+  design_data$test <- test
+  design_data$reduced_design <- reduced_design
+  label <- paste0("test=", test %||% "NULL/default (Wald)", ", type=", type %||% "NULL (Skip)")
+  dds <- make_dds(design_data,
+                  config=config,
+                  featureCounts='featurecounts.txt',
+                  parallel=config$parallel$parallel)
+  tmp_dds_list = list(dds=dds)
+  res <- make_results(dds_name='dds',
+                      label=label,
+                      dds_list=tmp_dds_list,
+                      type=type)
+  return(list(dds=dds, res=res))
+}
+
+# Create objects for testing defaults
+dds_and_res <- make_deseq_results()
+dds <- dds_and_res$dds
+res <- dds_and_res$res$res
+
+wald_dds_list <- list(dds1=dds)
+wald_res_list <- list(res1=list(res=res, dds='dds1', label='Defaults'))
+wald_res_list <- lcdbwf:::attach_extra(wald_res_list, config)
+
+# Create objects for testing 'LRT'
+dds_and_res <- make_deseq_results(test='LRT', type=NULL, reduced_design=~1)
+dds <- dds_and_res$dds
+res <- dds_and_res$res$res
+
+lrt_dds_list <- list(dds1=dds)
+lrt_res_list <- list(res1=list(res=res, dds='dds1', label='LRT'))
+lrt_res_list <- lcdbwf:::attach_extra(lrt_res_list, config)
+
+# ------ Test build_results_tabs function ------ #
+test_that("build_results_tabs works with Wald test", {
```

Comment:
Were there previous issues where this wasn't working? Or maybe LRT wasn't working, and that needed a test, and it makes sense to have a symmetrical test for Wald?

Reply:
I made a separate test for Wald for completions sake but there is no reason build_results_tabs should fail if the results object in res_list was created without error for LRT or Wald